### PR TITLE
Make test coverage 100%

### PR DESCRIPTION
**Pull request by 921kiyo**
## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

## Motivation and Context
Why was this PR created?

Enabled the test coverage check and added additional tests to make it 100%

## How has this been tested?
What testing strategies have you used?

Unit tests

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes
- [x] Assigned myself to the PR
- [ ] Added `Type` label to the PR

### CODE REVIEW

File: package/tests/test_server.py [hunk 1]
```diff
@@ -184,38 +184,64 @@ def test_no_browser(cli_runner):
...
+    """
+    Check that top level keys are properly checked.
+    """
+    filepath_json = str(tmp_path / "test.json")
+    data = {}
+    with open(filepath_json, "w") as f:
+        json.dump(data, f)
+
+    result = cli_runner.invoke(server.commands, ["viz", "--load-file", filepath_json])
+    assert result.output == "Invalid file, top level key 'nodes' not found.\n"
```
Comment (ZainPatelQB) [inline package/tests/test_server.py]: Can the top level keys be anything? If not, maybe add a (parametrized) test to see if `data = {"fake": "fake"}` also gives an `invalid file`?

File: package/tests/test_server.py [hunk 2]
```diff
@@ -184,38 +184,64 @@ def test_no_browser(cli_runner):
...
+    Check that running with `--save-file` flag saves pipeline JSON file in a specified path.
+    """
+    save_path = str(tmp_path / "test.json")
+
+    result = cli_runner.invoke(server.commands, ["viz", "--save-file", save_path])
     assert result.exit_code == 0, result.output
 
+    with open(save_path, "r") as f:
+        json_data = json.load(f)
+        assert json_data == EXPECTED_PIPELINE_DATA
```
Comment (ZainPatelQB) [inline package/tests/test_server.py]: The `assert` doesn't have to be indented, right? But I'm not fussed 🤷‍♂

File: package/kedro_viz/server.py [hunk 3]
```diff
@@ -71,7 +71,11 @@ def _hash(value):
 
 def _check_viz_up(port):
     url = "http://127.0.0.1:{}/".format(port)
-    response = requests.get(url)
+    try:
+        response = requests.get(url)
+    except requests.ConnectionError:
+        return False
+
```
Comment (ZainPatelQB) [inline package/kedro_viz/server.py]: Nice one! 👏

File: package/test_requirements.txt [hunk 4]
```diff
@@ -6,5 +6,6 @@ pytest-coverage
 behave>=1.2.6, <2.0
 pylint>=2.3.1, <2.4.0 # 2.4.1 doesn't work for Python 3.5, and requires investigation.
 pytest-mock>=1.7.1,<2.0
+requests-mock>=1.6.0, <2.0.0
```
Comment (ZainPatelQB) [inline package/test_requirements.txt]: To be consistent, maybe

```suggestion
requests-mock>=1.6.0, <2.0
```

I can see `pytest-mock>=1.7.1,<2.0` and `behave>=1.2.6, <2.0` for example.

File: package/tests/test_server.py [hunk 5]
```diff
@@ -184,38 +184,64 @@ def test_no_browser(cli_runner):
...
 
+    with open(save_path, "r") as f:
+        json_data = json.load(f)
+    assert json_data == EXPECTED_PIPELINE_DATA
+
+
+def test_no_top_level_key(cli_runner, tmp_path):
+    """
+    Check that top level keys are properly checked.
+    """
```
Comment (lorenabalan) [inline package/tests/test_server.py]: I'd put all of these one-liner docstrings on one line, think that's more consistent with Google docstrings.
`"""blabla"""`

File: package/tests/test_server.py [hunk 6]
```diff
@@ -289,3 +315,18 @@ def test_call_twice_with_different_port(self, mocked_process):
...
+        Test the helper function which checks if HTTP GET status code is 200.
+        """
+        requests_mock.get(
+            "http://127.0.0.1:8000/", content=b"some output", status_code=200
+        )
+        assert server._check_viz_up(8000)  # pylint: disable=protected-access
+
+    def test_check_viz_up_invalid(self):
+        """
+        Test should catch the request connection error and returns false.
```
Comment (lorenabalan) [inline package/tests/test_server.py]:
```suggestion
        Test should catch the request connection error and returns False.
```
I'd rename the tests to be more explicit that one checks the status_code and one the connection error scenario. "invalid" is a bit vague.

File: package/tests/test_server.py [hunk 7]
```diff
@@ -184,38 +184,64 @@ def test_no_browser(cli_runner):
...
+
+    result = cli_runner.invoke(server.commands, ["viz", "--save-file", save_path])
     assert result.exit_code == 0, result.output
 
+    with open(save_path, "r") as f:
+        json_data = json.load(f)
+    assert json_data == EXPECTED_PIPELINE_DATA
+
+
+def test_no_top_level_key(cli_runner, tmp_path):
```
Comment (lorenabalan) [inline package/tests/test_server.py]:
```suggestion
def test_load_file_no_top_level_key(cli_runner, tmp_path):
```

**Pull request merged**